### ISSUE TITLE
Receiver is never unregistered

### ISSUE THREAD

**Issue by sir-boformer** [esafirm/RxDownloader#17]
``RxDownloader`` calls ``context.registerReceiver``, but the receiver is never unregistered after the download is complete.

**Comment by esafirm** [comment]
@sir-boformer you're right. I think we can change the `registerReceiver` part when user start download and unregister it once the map that track download subject is empty.

Any suggestion?

**Comment by shishirshetty0710** [comment]
@esafirm still getting this error:
**MainActivity has leaked IntentReceiver com.esafirm.rxdownloader.RxDownloader$DownloadStatusReceiver@af89719 that was originally registered here. Are you missing a call to unregisterReceiver()?**

**Comment by kojofosu** [comment]
@esafirm  Getting this error too.. will it be fixed soon?

**E/ActivityThread: Activity  has leaked IntentReceiver com.esafirm.rxdownloader.RxDownloader$DownloadStatusReceiver@544c389 that was originally registered here. Are you missing a call to unregisterReceiver()?**
